examples/genetic_music/sketch.js

```diff
@@ -0,0 +1,240 @@
+  }
+}
+
+function draw() {
+  background(30);
+  for (var i=0; i<population.length; i++) {
+    population[i].display();
+  }
+  fill(255);
+  if (songIsPlaying) {
+    text("Song playing... Click to stop.", width/2, height/2);
+  } else {
+    text("Click on an Earworm to hear it sing!", width/2, height/2);
+  }
+  text("Generation: " + generationCount, width/2, height/6);
+}
+
+function mousePressed() {
+  if (songIsPlaying) {
+    // Stop a song
+    sloop.stop();
+    songIsPlaying = false;
+  } else {
+    // Start a song
+    for (var i=0; i<population.length; i++) {
+      var clickToEarwormDistance = dist(mouseX, mouseY, population[i].xpos, population[i].ypos);
+      if (clickToEarwormDistance < population[i].radius) {
+        clickedEarwormIndex = i;
+        notePlaybackIndex = 0;
+        songIsPlaying = true;
+        sloop.start();
+      }
+    }  
+  }
+}
+
+function selectFittest() {
+  // Sort in descending order of fitness
+  population.sort((a, b) => b.fitnessScore - a.fitnessScore);
+  // Keep only the N fittest
+  population = subset(population, 0, numberOfSurvivors);
+  // Re-assign ID numbers
+  for (var i=0; i<population.length; i++) {
+    population[i].id = i;
+  }
+}
+
+function reproducePopulation() {
+  var newPopulation = [];
+  while (newPopulation.length < maxPopulationSize - numberOfSurvivors) {
+    var parentA = random(population);
+    var parentB = random(population);
+    var child = parentA.reproduceWith(parentB);
+    newPopulation.push(child);
+  }
+  // Add new generation to the survivors
+  population = population.concat(newPopulation);
+  // Re-assign ID numbers
+  for (var i=0; i<population.length; i++) {
+    population[i].id = i;
+  }
+  generationCount++;
+}
+
+function fastForward() {
+  var fastForwardNum = 10;
+  for (var i=0; i<fastForwardNum; i++) {
+    selectFittest();
+    reproducePopulation();
+  }
+}
+
+function resetPopulation() {
+  generationCount = 1;
+  for (var i=0; i<maxPopulationSize; i++) {
+    var song = new Earworm(i);
+    song.initialize();
+    population[i] = song;
+  }
+}
+
+function Earworm(indexNumber) {
+  this.id = indexNumber;
+  this.length = songLength;
+  this.notes = [];
+  // Visual properties
+  this.xpos = random(width);
+  this.ypos = random(height);
+  this.radius = (width + height) / 50;
+  this.fitnessScore = 0;
+}
+Earworm.prototype.initialize = function() {
+  this.notes = [];
+  for (var i=0; i<this.length; i++) {
+    this.notes.push(random(validNotes));
+  }
+  this.calculateFitness();
+};
+Earworm.prototype.display = function() {
+  this.xpos = constrain(this.xpos + random(-1, 1), 0, width);
+  this.ypos = constrain(this.ypos + random(-1, 1), 0, height);
+
+  push();
+  strokeWeight(1);
+  angleMode(DEGREES); // Change the mode to DEGREES
+  var angle = 360 / this.notes.length;
+  translate(this.xpos, this.ypos);
+  for (var i=0; i<this.notes.length; i++) {
+    var color = map(this.notes[i], minValidNote, maxValidNote, 280, 120) % 255;
```

Comment:
if color corresponded to pitch class (0 - 11) that would help show patterns across octaves